Fix -Wimplicit-function-declaration error with gcc 14.

Error:

```
./test/test_lurch_util.c: In function 'test_lurch_util_axc_log_func_error':
./test/test_lurch_util.c:69:5: error: implicit declaration of function
'lurch_util_axc_log_func' [-Wimplicit-function-declaration]
   69 |     lurch_util_axc_log_func(AXC_LOG_ERROR, "test", 4, axc_ctx_p);
      |     ^~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/lurch_util.h
+++ b/src/lurch_util.h
@@ -21,6 +21,16 @@
 int lurch_util_axc_get_init_ctx(char * uname, axc_context ** ctx_pp);
 
 /**
+ * Log wrapper for AXC
+ *
+ * @param level an AXC_LOG level
+ * @param msg   the log message
+ * @param len the length of the message
+ * @param ctx_p the axc context
+ */
+void lurch_util_axc_log_func(int level, const char * msg, size_t len, void * user_data);
+
+/**
  * For some reason pidgin returns account names with a trailing "/".
  * This function removes it.
  * All other functions asking for the username assume the "/" is already stripped.
